### ISSUE TITLE
Catch exception in post

### ISSUE THREAD

**Issue by bartv** [inmanta/inmanta#691]
An exception in process_events results in a result that is not reported back to the server. post method is also called before process_events

**Comment by bartv** [comment]
```
2018-07-03 17:58:27,454 ERROR   Future <tornado.concurrent.Future object at 0x7f22acacca90> exception was never retrieved: Traceback (most recent call last):
  File "/opt/inmanta/lib/python3.6/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/inmanta/lib/python3.6/site-packages/inmanta/agent/agent.py", line 180, in execute
    success, send_event = yield self._execute(ctx=ctx, events=received_events, cache=cache)
  File "/opt/inmanta/lib/python3.6/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/opt/inmanta/lib/python3.6/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "/opt/inmanta/lib/python3.6/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/inmanta/lib/python3.6/site-packages/inmanta/agent/agent.py", line 126, in _execute
    yield self.scheduler.agent.thread_pool.submit(provider.process_events, ctx, self.resource, events)
  File "/opt/inmanta/lib/python3.6/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/usr/lib64/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/usr/lib64/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/lib64/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/var/lib/inmanta/e8042189-f643-4712-924f-ea1b6f37203b/agent/code/modules/inmanta_plugins.fortimanager.py", line 244, in process_events
    self._call(ctx, resource)
  File "/var/lib/inmanta/e8042189-f643-4712-924f-ea1b6f37203b/agent/code/modules/inmanta_plugins.fortimanager.py", line 254, in _call
    raise Exception("Unable to execute")
Exception: Unable to execute
```

**Comment by arnaudsjs** [comment]
The following issue is resolved: "An exception in process_events results in a result that is not reported back to the server. "

The issue regarding the execution order of the post and the process_events method will be addressed in #1059
